TombEngine/Objects/Generic/puzzles_keys.cpp

```diff
@@ -55,6 +55,20 @@ const ObjectCollisionBounds KeyHoleBounds =
 		EulerAngles(ANGLE(10.0f), ANGLE(30.0f), ANGLE(10.0f)))
 };
 
+const auto WaterKeyHolePosition = Vector3i(0, -571, 0);
```

Comment:
Where this `-571` number comes from?

Reply:
This is the standard height of all puzzle and key items. I have added a comment.